ui/widgets/nuxeo-collapsible.js

```diff
@@ -0,0 +1,171 @@
+            width: 100%;
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            margin: 0;
+            padding: var(--nuxeo-collapsible-heading-padding, 4px);
+            background: transparent;
+            border: none;
+            cursor: pointer;
+            text-decoration: none;
+            font-size: 1rem;
+          }
+          .heading {
+            width: calc(100% - var(--nuxeo-collapsible-icon-width, 20px));
+            overflow-x: hidden;
+            text-align: start;
+          }
+          iron-icon {
+            --iron-icon-height: var(--nuxeo-collapsible-icon-height, 20px);
+            --iron-icon-width: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          .start {
+            flex-direction: row-reverse;
+            justify-content: flex-end;
+          }
+
+          .start-alignment {
+            margin-inline-start: var(--nuxeo-collapsible-icon-width, 20px);
+          }
+          iron-collapse {
+            padding: 0 var(--nuxeo-collapsible-heading-padding, 4px);
+          }
+        </style>
+
+        <button aria-expanded="[[expanded]]" on-tap="_toggle" class$="[[_changeChevronPosition(chevronAtStart)]]">
+          <slot class="heading" name="heading"></slot>
+          <iron-icon icon="[[_toggleIcon(expanded)]]" toggle hidden$="[[!chevron]]"></iron-icon>
+        </button>
+        <iron-collapse
+          opened="[[expanded]]"
+          no-animation="[[!animation]]"
+          class$="[[_alignCollapsedContent(chevronAtStart)]]"
+        >
+          <slot></slot>
+        </iron-collapse>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-collapsible';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * Set animation to false to disable animations
+         */
+        animation: {
+          type: Boolean,
+          value: true,
+        },
+
+        /**
+         * Show or hide the chevron icon
+         */
+        chevron: {
+          type: Boolean,
+          value: false,
+        },
+
+        /**
+         * Chevron icon position. By default it is positioned at the end of the heading,
+         * but it can be swapped to the its beginning
+         */
+        chevronAtStart: {
+          type: Boolean,
+          value: false,
+        },
+
+        /**
+         * Expanded or collapsed by default
+         */
+        expanded: {
+          type: Boolean,
+          value: false,
+          reflectToAttribute: true,
+        },
+      };
+    }
+
+    _changeChevronPosition() {
+      if (this.chevron && this.chevronAtStart) {
+        return 'start';
+      }
+    }
+
+    _alignCollapsedContent() {
+      if (this.chevron && this.chevronAtStart) {
+        return 'start-alignment';
+      }
+    }
+
+    _toggle() {
+      this.expanded = !this.expanded;
+    }
+
+    _toggleIcon(expanded) {
+      return `hardware:keyboard-arrow-${expanded ? 'up' : 'down'}`;
```

Comment:
I'm assuming we don't want the icon to be configurable.

Reply:
Yes, for now we will force the usage of this icon for consistency sake.